android/library/src/main/java/com/liferay/mobile/screens/thingscreenlet/screens/views/row/WorkflowTaskRowView.kt

```diff
@@ -0,0 +1,103 @@
+import com.liferay.apio.consumer.extensions.shortFormat
+import com.liferay.apio.consumer.model.Thing
+import com.liferay.mobile.screens.R
+import com.liferay.mobile.screens.thingscreenlet.delegates.bindNonNull
+import com.liferay.mobile.screens.thingscreenlet.model.WorkflowTask
+import com.liferay.mobile.screens.thingscreenlet.screens.ThingScreenlet
+import com.liferay.mobile.screens.thingscreenlet.screens.views.BaseView
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+
+class WorkflowTaskRowView @JvmOverloads constructor(
+    context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0, defStyleRes: Int = 0) : BaseView,
+    LinearLayout(context, attrs, defStyleAttr) {
+
+    override var screenlet: ThingScreenlet? = null
+
+    private val isPending by bindNonNull<FrameLayout>(R.id.workflow_is_pending)
+    private val assetTitle by bindNonNull<TextView>(R.id.workflow_asset_title)
+    private val dateCreated by bindNonNull<TextView>(R.id.workflow_date_created)
+    private val type by bindNonNull<TextView>(R.id.workflow_type)
+    private val status by bindNonNull<TextView>(R.id.workflow_status)
+
+    override var thing: Thing? by converter<WorkflowTask> {
+
+        if (it.completed) isPending.visibility = View.INVISIBLE
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            assetTitle.text = Html.fromHtml(getAssetTitle(it), Html.FROM_HTML_MODE_COMPACT)
+        } else {
+            assetTitle.text = Html.fromHtml(getAssetTitle(it))
+        }
+
+        dateCreated.text = if (it.dateCreated != null && isYesterday(it.dateCreated))
+            resources.getString(R.string.workflow_yesterday_text)
+        else it.dateCreated?.shortFormat(Locale.getDefault())
+
+        type.text = resources.getString(R.string.workflow_type_text, it.name.capitalize(), getAssetName(it))
+
+        when {
+            it.completed -> status.text = resources.getString(R.string.workflow_completed_text)
+            it.expires != null -> {
+                val days = getDays(it.expires)
+
+                if (days < 0) status.text = resources.getQuantityString(R.plurals.workflow_due_date_before_text,
+                    Math.abs(days), Math.abs(days))
+                else status.text = resources.getQuantityString(R.plurals.workflow_due_date_after_text, days, days)
+            }
+            else -> status.text = resources.getString(R.string.workflow_pending_text)
+        }
+    }
+
+    private fun getAssetName(workflowTask: WorkflowTask): String {
+        if (workflowTask.comment != null) return workflowTask.comment.type!!
+        if (workflowTask.blogPost != null) return workflowTask.blogPost.type!!
+        return "Undefined"
+    }
+
+    private fun getAssetTitle(workflowTask: WorkflowTask): String {
+        if (workflowTask.comment != null) return workflowTask.comment.text!!
+        if (workflowTask.blogPost != null) return workflowTask.blogPost.headline!!
+        return "Undefined"
```

Comment:
what do you think of: `workflowTask.comment?.text ?: workflowTask.blogPost?.headline ?: "Undefined" `

Reply:
This is going to change anyway, but it's better your approach. I'll change it, thanks!